reframe/frontend/cli.py

```diff
@@ -877,7 +912,7 @@ def print_infoline(param, value):
         'cmdline': ' '.join(sys.argv),
         'config_file': rt.site_config.filename,
         'data_version': runreport.DATA_VERSION,
-        'hostname': socket.getfqdn(),
+        'hostname': get_hostname_cmd(autodetect_opts),
```

Comment:
I would probably use here `socket.gethostname()` and add some debug messages in the autodetection process, since the hostname here is just informational.